src/gmt_api.c

```diff
@@ -12739,12 +12746,27 @@ int GMT_Put_Vector (void *API, struct GMT_VECTOR *V, unsigned int col, unsigned
 		case GMT_SHORT:		V->type[col] = GMT_SHORT;	V->data[col].si2 = vector;	break;
 		case GMT_UCHAR:		V->type[col] = GMT_UCHAR;	V->data[col].uc1 = vector;	break;
 		case GMT_CHAR:		V->type[col] = GMT_CHAR;	V->data[col].sc1 = vector;	break;
+		case GMT_DATETIME:	/* Must convert from string-time to double */
+			if ((dt = gmtapi_get_char_char_ptr (vector)) == NULL) {
+				GMT_Report (API, GMT_MSG_ERROR, "Datetime string array is NULL\n");
+				return GMT_MEMORY_ERROR;
+			}
+			if ((t_vector = malloc (V->n_rows * sizeof(double))) == NULL) {
+				GMT_Report (API, GMT_MSG_ERROR, "Unable to allocate array of %" PRIu64 " doubles for converted datetime strings\n", V->n_rows);
+				return GMT_MEMORY_ERROR;
+			}
+			for (row = 0; row < V->n_rows; row++) {
+				if (gmt_scanf_argtime (API->GMT, dt[row], &(t_vector[row])) == GMT_IS_NAN) n_bad++;
```

Comment:
It now crashes when accessing the value of `dt[row]`.

Reply:
So this is what the code does:
`char **dt = NULL;  /* Pointer to array of strings in C */`
`dt = gmtapi_get_char_char_ptr (vector); `
This is just a convenience function that accepts the void *vector pointer and casts it as a char ** pointer.  If we do (char **) instead we get nasty warnings but it is basically the same thing
So now, `dt[row]` should get your strings.  However, I remember Leo talking about C-access to numpy arrays so perhaps there is a special scheme to pass whatever string array you have to a C function that expects an array of strings?

Reply:
OK, need more debugging from the Python side.

Reply:
Maybe [this](https://stackoverflow.com/questions/3494598/passing-a-list-of-strings-to-from-python-ctypes-to-c-function-expecting-char)?

Reply:
This branch works well for datetime `2006-01-01T`, `2006-01-01T05:00` or similar, but doesn't work for `2006-01-01` with the following error:
```
pygmt-session [ERROR]: Unable to parse 2datetime strings (ISO datetime format required)
```
The `T` is not required for GMT command line.

Reply:
Are you sure? What does the ISO convention demand.  Well, I guess we have to append a T internally.  Will see after dinner.

Reply:
OK, try now.

Reply:
It works now. But GMT still reports the error:
```
Unable to parse 3 datetime strings (ISO datetime format required).
```
Perhaps something wrong with the `if(gmt_scanf())` test?